### PR TITLE
many: avoid snap.InfoFromSnapYaml in tests

### PR DESCRIPTION
**Pull request by zyga**
This branch switches all of the interface/ code from `snap.InfoFromSnapYaml`, which does no validation, to the new `snaptest.MockInfo` (similar to the existing MockSnap). This ensures that YAML data in tests is accurately testing how the system would behave in a given situation by preventing invalid YAML to silently cause issues.

### CODE REVIEW

File: snap/snaptest/snaptest.go [hunk 1]
```diff
@@ -55,6 +55,23 @@ func MockSnap(c *check.C, yamlText string, sideInfo *snap.SideInfo) *snap.Info {
 	return snapInfo
 }
 
+// MockInfo parses snap.yaml and returns a validated snap.Info object.
```
Comment (pedronis) [inline snap/snaptest/snaptest.go]: ... parses the given snap.yaml text and returns ... object including the optional sideInfo.

**Comment by pedronis** [comment]
picking this up quickly in #1966

**Comment by pedronis** [comment]
thanks @zyga, opened #1966 to address small the doc comment tweak

File: snap/snaptest/snaptest.go (hunk 1 above)
Comment (zyga) [inline snap/snaptest/snaptest.go]: Thanks for fixing this in your branch :-)

**Pull request closed without merging**